command.go

```diff
@@ -1004,10 +1004,14 @@ func xMessageSliceParser(rd *proto.Reader, n int64) (interface{}, error) {
 			}
 
 			v, err := rd.ReadArrayReply(stringInterfaceMapParser)
-			if err != nil {
+			if err != nil && err != proto.Nil {
 				return nil, err
 			}
 
+			if v == nil || err == proto.Nil {
+				v = make(map[string]interface{})
```

Comment:
Is there a reason to create an empty map? `nil` is a natural choice, saves 1 allocation, and you still can read from nil map - e.g. `v["foo"]` works just fine.

Reply:
I guess you did it because of `v.(map[string]interface{})` few line below. I will change this myself but let me know if it breaks anything for you.

Reply:
yeah, I did that because of the `v.(map[string]interface{})` below. You can change it to `nil` if you think `nil` is more natural here.

Reply:
Thanks, I've done just that in v6 & v7.